src/data/matrix/basic.lean

```diff
@@ -437,6 +453,75 @@ lemma vec_mul_vec_eq (w : m → α) (v : n → α) :
   vec_mul_vec w v = (col w) ⬝ (row v) :=
 by { ext i j, simp [vec_mul_vec, mul_val], refl }
 
+variables [decidable_eq m] [decidable_eq n]
+
+/--
+`std_basis_matrix i j a` is the matrix with `a` in the `i`-th row, `j`-th column,
+and zeroes elsewhere.
+-/
+def std_basis_matrix (i : m) (j : n) (a : α) : matrix m n α :=
+(λ i' j', if i' = i ∧ j' = j then a else 0)
+
+@[simp] lemma smul_std_basis_matrix (i : m) (j : n) (a b : α) :
+b • std_basis_matrix i j a = std_basis_matrix i j (b • a) :=
+by { unfold std_basis_matrix, ext, dsimp, simp }
+
+@[simp] lemma std_basis_matrix_zero (i : m) (j : n) :
+std_basis_matrix i j (0 : α) = 0 :=
+by { unfold std_basis_matrix, ext, simp }
+
+lemma std_basis_matrix_add (i : m) (j : n) (a b : α) :
+std_basis_matrix i j (a + b) = std_basis_matrix i j a + std_basis_matrix i j b :=
+begin
+  unfold std_basis_matrix, ext,
+  split_ifs with h; simp [h],
+end
+
+
+lemma matrix_eq_sum_elementary (x : matrix n m α) :
+x = ∑ (i : n) (j : m), std_basis_matrix i j (x i j) :=
+begin
+  ext, iterate 2 {rw finset.sum_apply},
+  rw ← finset.sum_subset, swap 4, exact {i},
+  { norm_num [std_basis_matrix] },
+  { simp },
+  intros, norm_num at a, norm_num,
+  convert finset.sum_const_zero,
+  ext, norm_num [std_basis_matrix],
+  rw if_neg, tauto!,
+end
+
+lemma elementary_eq_basis_mul_basis (i : m) (j : n) :
```

Comment:
```suggestion
-- todo: tie this up with the `basis` machinery of linear algebra
-- this is not completely trivial because we are indexing by two types, instead of one

-- todo: add `std_basis_vec`
lemma elementary_eq_basis_mul_basis (i : m) (j : n) :
```